apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette-content-type/dot-palette-content-type.component.html

```diff
@@ -0,0 +1,25 @@
+<dot-palette-input-filter
+    #filterInput
+    (filter)="filterContentTypes($event)"
+></dot-palette-input-filter>
+<div
+    class="dot-content-palette__items"
+    *ngIf="itemsFiltered === undefined || itemsFiltered?.length; else noResults"
```

Comment:
Why this condition `itemsFiltered === undefined`? Shouldn't it be `itemsFiltered !== undefined`?

Reply:
yes, your logic sounds fine, but on this case on the IF
`*ngIf="itemsFiltered === undefined || itemsFiltered?.length; else noResults"`

it does not apply, if I put your change then`noResults` will be shown the first time the component is loaded at the very first moment (because the data is being loaded from the BE), and then when the data is loaded it will be shown the list of elements

so it creates a undesired look and feel flickering behaviour, showing first the NO Results view, and then switching to showing the list

Reply:
I see, then it is the correct condition. However, isn't an error being thrown? ? Because at first the value is undefined and you have an `*ngFor` try to iterate an undefined one.

Reference: https://github.com/dotCMS/core-web/blob/5d756c5a0898b05d5edce9f9d25d6d0b2572a5f0/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette-content-type/dot-palette-content-type.component.html#L9-L14

Note: If you are not getting errors there would be no problems.